test/eventHandlers/InvocationHandler.test.ts

```diff
@@ -1063,4 +1063,37 @@ describe('InvocationHandler', () => {
             msg.invocation.response([])
         );
     });
+
+    it('log hook respects changes to value, only for user log', async () => {
+        coreApi.registerHook('log', (ctx) => {
+            ctx.message += 'UpdatedFromHook';
+            ctx.level = 'error';
+        });
+
+        registerV3Func(Binding.queue, async (invocContext: Context) => {

```

Comment:
General question. I see the worker tests have this method, but why is it called `registerV3Func`? I don't see a `registerV4Func` anywhere. Isn't it just registering _a_ function? Why the distinction between V3 and V4?

Reply:
v3 is special because it's the default if you don't have a programming model specified. The worker has a [reference](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/package.json#L8) to the v3 package because of this, but not the v4 package. Also most of these tests were written before the v3/v4 split, so they're using v3 for historical reasons

We have a few tests to verify the core api (which the v4 model uses to register a function), but in general the worker shouldn't need to know anything about the programming model so it's okay that we're not explicitly testing v4 in this repo